### DISCUSSION TITLE
Ability to Use Other Laravel Packages with Laravel Zero?

### DISCUSSION THREAD

**Discussion by orware** [laravel-zero/laravel-zero#340] [Q&A]
Good evening Nuno,

I wanted to ask...aside from the supported packages you have available for Laravel Zero, should it be expected for there to be much of an issue trying to utilize other Laravel Packages with Laravel Zero?

I was initially trying out Laravel Zero to create a set of new CLI scripts to replace some aging ones I need to update and move to something more modern, and Laravel Zero seemed like a good option since it seemed to be more tailored for something that only needed the CLI support.

However, when I added in the package I need to use for Oracle database connectivity (https://github.com/yajra/laravel-oci8) I wasn't able to run the vendor publish command it has to bring it's oracle.php config file over...so I brought it over manually to the config folder, and that still didn't work (it didn't appear that Laravel Zero was including that config file). I then brought over the connection into the database.php config file and that seemed to help slightly (as things moved to a different error).

I added their Service Provider class in manually as well to my app.php config and at that point received a `Target class [auth] does not exist.` error and that's when I went ahead and switched over to a regular Laravel install instead (at least for the moment) and was able to install the Oracle package / publish its config / run a quick test query via a new Command I created without any issues.

I'm not sure if most of the existing usage of Laravel Zero by yourself and others has been fortunate enough not to run into this issue yet, but hopefully there may be a way to allow packages like the Oracle related one above to work more seamlessly with Laravel Zero since I do see how it would be a bit more streamlined to utilize instead of a full regular Laravel install for this particular situation.

Thank you!

**Comment by nunomaduro** [comment]
cc @owenvoke

**Reply by owenvoke** [reply]
I've found that a lot of packages do work with Laravel Zero. However, as it's designed to be a lightweight command-line framework, we don't include a lot of the Illuminate packages (such as `illuminate/auth`).

Most of the time, you can just add packages manually and they will work out of the box (although the service provider needs to be registered manually).

I don't seem to be able to get the OCI8 extension working on my test environments, so I'm not able to check the compatibility for this specific package, however you might be able to just require `illuminate/auth` and then register the `Illuminate\Foundation\Support\Providers\AuthServiceProvider::class` in your `config/app.php` file.
